### ISSUE TITLE
NamedArguments: False positive on spread varargs

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#7272]
## Expected Behavior
No flagging on varargs (unless maybe there's 3 or 4 other nameable arguments).

## Observed Behavior
> This function call has 4 arguments. To call a function with more than 3 arguments you should set the name of each argument. [NamedArguments]

## Steps to Reproduce
```kotlin
fun varargFun(str: String, vararg nums: Int) { TODO(str + nums) }

fun call() {
	varargFun("a", 1, 2, 3, 4, 5) // this is fine (no NamedArguments)
	val nums1 = intArrayOf(1, 2, 3) // simulate
	val nums2 = intArrayOf(4, 5, 6) // complex
	val nums3 = intArrayOf(7, 8, 9) // logic
	varargFun("a", *nums1, *nums2, *nums3) // this is not (unexpected NamedArguments)
}
```

## Context
It's impossible to add named arguments in this case, the best we can do is `varargFun(str = "a", ...)` but even that is flagged.

## Your Environment

* Version of detekt used: 1.23.6
* Version of Gradle used (if applicable): 8.7
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): N/A

**Comment by hjed** [comment]
Any chance this fix could be backported to 1.x.x ?  
It breaks things like `listOf()` which makes this rule a bit useless.
